src/imitation/algorithms/preference_comparisons.py

```diff
@@ -676,6 +683,8 @@ def __init__(
                 in returns that are above this threshold. This threshold
                 is therefore in logspace. The default value of 50 means
                 that probabilities below 2e-22 are rounded up to 2e-22.
+            weight_decay: the weight decay factor for the reward model's weights,
+                equivalent to L2 regularization.
```

Comment:
To be pedantic, weight decay and L2 regularization are _not_ equivalent for Adam (see https://arxiv.org/abs/1711.05101). `torch.optim.Adam` uses L2, whereas `torch.optim.AdamW` uses "real" weight decay. I don't think we need to discuss this in detail, but maybe write something like "weight decay factor (implemented as L2 regularization rather than AdamW, see https://arxiv.org/abs/1711.05101)" to avoid confusion?

Reply:
Should we just move to `AdamW` by default? From the linked paper it seems preferable, as it decouples the choice of this hyperparameter from the learning rate.

Reply:
I don't have any practical experience with `Adam` vs `AdamW` (haven't even read the entire paper). Don't think it matters much, but I'd lean slightly towards switching to `AdamW` (I don't think it has overtaken `Adam` as the "standard" optimizer, but I do think it has seen substantial adoption, so I'd guess they decoupling advantages are real?)